var/spack/repos/builtin/packages/stream/package.py

```diff
@@ -25,9 +26,21 @@ def edit(self, spec, prefix):
         makefile.filter('CC = .*', 'CC = cc')
         makefile.filter('FC = .*', 'FC = f77')
 
-        cflags = '-O2'
-        fflags = '-O2'
-        if '+openmp' in self.spec:
+        if self.compiler.name == 'aocc':
+            omp_flag = self.compiler.openmp_flag
```

Comment:
Here you are adding the openmp flag regardless of whether `+openmp` or `~openmp` is requested. I think you should remove the openmp logic from this block and keep it in the subsequent block for all compilers.